src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_grand_champions.cpp

```diff
@@ -15,577 +15,1264 @@
+                            pGrandChampion->AI()->AttackStart(plr);
+                    }
+                }
+                break;
             }
         }
+    }
+
+    bool CanUseNormalAI()
+    {
+        // must check for unattackable flag to prevent crash in certain situations
+        return !UpdateVictim() || me->GetVehicleBase() || LookingForMount || me->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE) || !MountedPhaseDone;
+    }
+
+    void EnterEvadeMode(EvadeReason why = EVADE_REASON_OTHER) override
+    {
+        if (LookingForMount || !MountedPhaseDone)
+            return;
 
-        void JustEngagedWith(Unit* /*who*/) override
+        // we must check if other champions have been downed and call evade for them
+        // for example, if players wipe during killing the last champion
+        for (int i = 0; i < 3; ++i)
         {
-            DoCastSpellShield();
+            if (Creature* pGrandChampion = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_GRAND_CHAMPION_1 + i)))
+            {
+                if (pGrandChampion->HasAura(SPELL_KNEEL))
+                {
+                    pGrandChampion->RemoveAllAuras();
+                    pGrandChampion->GetThreatManager().ClearAllThreat();
+                    pGrandChampion->SetLootRecipient(nullptr);
+                    pGrandChampion->SetRegenerateHealth(true);
+                    pGrandChampion->GetMotionMaster()->MoveTargetedHome();
+                    pGrandChampion->AI()->Reset();
+                }
+            }
         }
+        BossAI::EnterEvadeMode(why);
+    }
 
-        void DoCastSpellShield()
+    void JustReachedHome() override
+    {
+        if (bHome)
         {
-            for (uint8 i = 0; i < 3; ++i)
-                DoCast(me, SPELL_SHIELD, true);
+            // Champions will be unattackable for 15 seconds when reached their homepositions for first time after jousting event
+            // After that time, they will start attacking a random player
+            // if group wipes during actual fight, grand champions will be attackable immediately when reached home position
+            uiPhase = 4;
+            events.ScheduleEvent(EVENT_START_ATTACK_REAL, 15000);
+            bHome = false;
         }
+        else if (MountedPhaseDone && !bHome)
+            EnterAggressiveMode(me);
+        _JustReachedHome();
+    }
 
-        void UpdateAI(uint32 uiDiff) override
+    void SpellHit(Unit* caster, SpellInfo const* spell) override
+    {
+        if (spell->Id == SPELL_TRAMPLE_AURA && LookingForMount && uiPhase == 0 && !me->IsImmunedToSpell(spell, caster))
         {
-            EscortAI::UpdateAI(uiDiff);
+            uiPhase = 3;
+            //me->GetMotionMaster()->MovementExpired();
+            me->GetMotionMaster()->Clear(MOTION_PRIORITY_NORMAL);
+            me->GetMotionMaster()->MoveIdle();
+            Talk(EMOTE_TRAMPLE, me);
+            events.ScheduleEvent(EVENT_TRAMPLE, spell->GetDuration());
+        }
```

Comment:
Are you certain the spell you should use isn't 67865 and its triggered spell 67866 ?